sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -102,10 +102,12 @@ case class HashAggregateExec(
 
       val beforeAgg = System.nanoTime()
       val hasInput = iter.hasNext
-      val res = if (!hasInput && groupingExpressions.nonEmpty) {
-        // This is a grouped aggregate and the input iterator is empty,
+      val res = if (!hasInput && (groupingExpressions.nonEmpty || resultExpressions.isEmpty)) {
+        // The input iterator is empty, and this is a grouped aggregate or without result columns,
         // so return an empty iterator.
         Iterator.empty
+      } else if (!hasInput && resultExpressions.isEmpty) {
```

Comment:
This is covered in the previous condition, isn't it?

Reply:
oops, forgot to remove it.